### PR TITLE
[parents_migration] Skip documents without parents in Confluence

### PR DESCRIPTION
**Pull request by aubin-tchoi**
## Description

- Some data_source_documents in Confluence have 0 parent.
- This seems to be some legacy issue.
- Unlike Slack we cannot recover the parents using only the `nodeId`.
- Plan is to not handle them for now (only log), the implement a `sync-page` in Confluence admin CLI and use it to fix the parents of the skipped pages.

## Deploy Plan

no deploy.

### CODE REVIEW

File: front/migrations/20241211_parents_migrator.ts [hunk 1]
```diff
@@ -21,10 +21,12 @@ const isMigratorAction = (action: string): action is MigratorAction => {
 };
 
 type ProviderMigrator = {
+  /// returns [nodeId, ...oldParents, ...newParents] idempotently
   transformer: (
     nodeId: string,
     parents: string[]
-  ) => { parents: string[]; parentId: string | null };
+  ) => { parents: string[]; parentId: string | null } | null; // null here means we skip the document
```
[the remark this answers is not in the document]
Comment (aubin-tchoi) [reply front/migrations/20241211_parents_migrator.ts:28]: hmm my concern was to make it faster

[the remark this answers is not in the document]
Comment (aubin-tchoi) [reply front/migrations/20241211_parents_migrator.ts:28]: actually that's not a valid reason

[the remark this answers is not in the document]
Comment (aubin-tchoi) [reply front/migrations/20241211_parents_migrator.ts:28]: I'll do what you suggest below 👍

File: front/migrations/20241211_parents_migrator.ts [hunk 2]
```diff
@@ -189,8 +191,15 @@ const migrators: Record<ConnectorProvider, ProviderMigrator | null> = {
...
-    transformer: (_, parents) => {
-      assert(parents.length > 1, "parents.length <= 1"); // the only documents are pages, they at least have the space as parent
+    transformer: (nodeId, parents) => {
+      if (parents.length <= 1) {
+        // we skip these for now
+        logger.warn(
+          { nodeId, parents, problem: "Not enough parents" },
+          "Invalid Confluence parents"
+        );
+        return null;
```
[the remark this answers is not in the document]
Comment (aubin-tchoi) [reply front/migrations/20241211_parents_migrator.ts:201]: SGTM

**Review by spolu — APPROVED** [review]

**Pull request merged**